### PR TITLE
TCP Timeout Test

### PR DESCRIPTION
**Pull request by macisasandwich**
Traffic server configured to timeout after 3 seconds. Test 1 delays returning any data for 5 seconds, therefore gets 504 response. Test 2 delays for 1 second and returns 204 response.

### CODE REVIEW

**Comment by scw00** [comment]
[approve ci]

**Comment by scw00** [comment]
@macisasandwich Please fix the autest !

**Comment by scw00** [comment]
[approve ci autest]

**Comment by scw00** [comment]
[approve ci autest]

**Comment by macisasandwich** [comment]
There were inconsistencies of tabs/spaces/newlines here and there. It's fixed now.

**Comment by scw00** [comment]
[approve ci]

**Comment by shinrich** [comment]
[approve ci autest]

**Comment by shinrich** [comment]
Looks like there were arbitrary whitespace changes in uWserver.py.  Are they necessary?  I'd hate to lose the effective change history on that file.

**Comment by maskit** [comment]
Please run `make autopep8`.

**Comment by dragon512** [comment]
[approve ci]

File: tests/gold_tests/timeout/ttfb.test.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+tr = Test.AddTestRun()
+tr.Processes.Default.Command = 'curl -i http://127.0.0.1:{0}/timeout'.format(ts.Variables.port)
+tr.Processes.Default.StartBefore(ts)
+tr.Processes.Default.StartBefore(server)
+tr.Processes.Default.Streams.stdout = "gold/timeout.gold"
+tr.Processes.Default.ReturnCode = 0
+
+tr = Test.AddTestRun()
+tr.Processes.Default.Command = 'curl -i http://127.0.0.1:{0}/no_timeout'.format(ts.Variables.port)
+# tr.Processes.Default.StartBefore(ts)
```
Comment (dragon512) [inline tests/gold_tests/timeout/ttfb.test.py]: I this needed or not?

**Comment by maskit** [comment]
[approve ci]

**Comment by maskit** [comment]
[approve ci]

File: tests/gold_tests/autest-site/trafficserver.test.ext [hunk 2]
```diff
@@ -121,7 +121,7 @@ def MakeATSProcess(obj, name, command='traffic_server', select_ports=True):
     # create subdirectories that need to exist (but are empty)
     # log directory has to be created with correct permissions
     p.Setup.MakeDir(log_dir)  # log directory has to be created
-    p.Setup.Chown(log_dir, "nobody", "nobody", ignore=True)
+    p.Setup.Chown(log_dir, "nobody", "nogroup", ignore=True)
```
Comment (dragon512) [inline tests/gold_tests/autest-site/trafficserver.test.ext]: I want to make sure we have the correct "value" for security being tested

@bryancall @SolidWallOfCode Is this a better value?

File: tests/gold_tests/timeout/timeout.test.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+Test.Summary = 'Testing ATS timeout'
+
+netns_name = Test.Name
+TIMEOUT = 3
+
+# Test.SkipIf(Condition.true("Skipping this test since running it requires superuser privilege, which introduces other problems"))
+
+# Windows admin checking temporarily using a deprecated API. When Autest upgrades on jenkins, we can switch to Condition.IsElevated
+if os.name == 'nt':
```
Comment (dragon512) [inline tests/gold_tests/timeout/timeout.test.py]: You don't need the windows code for this test.. please remove

**Comment by dragon512** [comment]
[approve ci]

File: tests/gold_tests/autest-site/trafficserver.test.ext (hunk 2 above)
Comment (macisasandwich) [reply tests/gold_tests/autest-site/trafficserver.test.ext]: on RHEL 7, the group name is nobody with gid 99. But on Fedora/Ubuntu, the corresponding group is nogroup with gid 65534. So I'm not sure what to put for the group field in chown. Calling chown twice to try both nobody and nogroup seems like a not-so-good solution.

**Comment by dragon512** [comment]
[approve ci]

File: tests/gold_tests/autest-site/trafficserver.test.ext (hunk 2 above)
Comment (shinrich) [reply tests/gold_tests/autest-site/trafficserver.test.ext]: You can grep through /etc/passwd to see which builtin user is present.

grep "^nobody:" /etc/passwd

Alternatively, does our CI system create an "ats" user for the tests to use @zwoop?

Comment (bryancall) [reply tests/gold_tests/autest-site/trafficserver.test.ext]: I would check what the default group is for nobody and use that.

**Comment by dragon512** [comment]
@macisasandwich can we check on the nobody issue above. rebase with latest fixes

**Comment by macisasandwich** [comment]
Will get on it soon.

**Pull request closed without merging**